rewrite-core/src/main/java/org/openrewrite/Preconditions.java

```diff
@@ -80,6 +80,11 @@ public static TreeVisitor<?, ExecutionContext> not(TreeVisitor<?, ExecutionConte
         return new TreeVisitor<Tree, ExecutionContext>() {
             @Override
             public Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+                SourceFile sourceFile = tree instanceof SourceFile ? (SourceFile) tree : null;
+                // calling `isAcceptable()` in case `v` overrides `visit(Tree, P)`
+                if (sourceFile != null && !v.isAcceptable(sourceFile, ctx)) {
+                    return SearchResult.found(tree);
```

Comment:
Why are we adding a search result?

Reply:
The idea here is that a visitor (used as a precondition) which is not applicable to some source file, should also evaluate to `true`.

In the following example we have a negated `PlainTextVisitor` precondition checking for `foo` in the content:

https://github.com/openrewrite/rewrite/blob/a438606dd0398802124e41767ef51c6f652363a4/rewrite-core/src/test/java/org/openrewrite/PreconditionsTest.java#L142-L162

The assumption here is that the (negated) precondition should evaluate to `true` for all text sources not containing `foo` and also any other source files (e.g. a quark). Does that make sense?

Reply:
Also note that the same search result marker is used as an indicator for a match after the `visit()` call (due to the negation the precondition must make some arbitrary change to indicate a match). So I chose to use the same technique here.

Reply:
Ah, sorry was missing the `not` because it was hidden in github review.